src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -397,7 +397,10 @@ private async Task GenerateContainerPathFromPatternAsync(AutoroutePart part)
                 _contentManager ??= _serviceProvider.GetRequiredService<IContentManager>();
 
                 var cultureAspect = await _contentManager.PopulateAspectAsync(part.ContentItem, new CultureAspect());
-                using (CultureScope.Create(cultureAspect.Culture))
+
+                var cultureOptions = _serviceProvider.GetService<IOptions<CultureOptions>>().Value;
+
+                using (CultureScope.Create(cultureAspect.Culture, useUserOverride: !cultureOptions.IgnoreSystemSettings))
```

Comment:
>  and let the logic be applied at the lowest level in CultureScope and the new extensions helpers.

Do you mean passing the `CultureOption` to the `CultureScope`?

Reply:
If for the options we don't use `UseUserOverride` but `IgnoreSystemSettings`, this to not be confused with the term `User` and to have a default value of `false`, what I meant is to use everywhere params named `IgnoreSystemSettings`.

So here in place of

    using (CultureScope.Create(cultureAspect.Culture, useUserOverride: !cultureOptions.IgnoreSystemSettings))

Would be

    using (CultureScope.Create(cultureAspect.Culture, ignoreSystemSettings: cultureOptions.IgnoreSystemSettings))

So I suggested to apply the logic, here I mean using the `!` operator, only at the lowest level where we use `new CultureInfo()`, here I mean for example in `CultureScope.Create()` and in the new extensions, here I'm reffering to `AddSupportedCultures()` and `AddSupportedUICultures()` that would have the same renamed param `ignoreSystemSettings` in place of `useUserOverride`.

Passing the whole `CultureOptions` is another good idea.

Reply:
I'm still prefering `UseSystemCulture` or `UserSelectedCultureSetttings` rather than `IgnoreSystemSettings`

> Passing the whole `CultureOptions` is another good idea.

I think passing boolean value is much much better, to make the API simple and easy for testing